include/odp/api/spec/comp.h

```diff
@@ -0,0 +1,819 @@
+ *  Independent hash-only operations are not supported. Implementation should
+ *  perform hash along with valid compression algo.
+ *  Macros, enums, types and operations to utilize compression.
+ *  @{
+ */
+
+/**
+ * @def ODP_COMP_SESSION_INVALID
+ * Invalid session handle
+ */
+
+/**
+ * @typedef odp_comp_session_t
+ * Compression/Decompression session handle
+ */
+
+/**
+ * @typedef odp_comp_compl_t
+ * Compression/Decompression completion event handle
+ */
+
+/**
+ * Compression API operation mode
+ */
+typedef enum {
+	/** Synchronous, return results immediately */
+	ODP_COMP_SYNC,
+	/** Asynchronous, return results via event queue */
+	ODP_COMP_ASYNC
+} odp_comp_op_mode_t;
+
+/**
+ * Comp API operation type.
+ *
+ */
+typedef enum {
+	/** Compress  */
+	ODP_COMP_OP_COMPRESS,
+	/** Decompress */
+	ODP_COMP_OP_DECOMPRESS
+} odp_comp_op_t;
+
+/**
+ * Comp API hash algorithm
+ *
+ */
+typedef enum {
+	/** ODP_COMP_HASH_ALG_NONE*/
+	ODP_COMP_HASH_ALG_NONE,
+	/** ODP_COMP_HASH_ALG_SHA1*/
+	ODP_COMP_HASH_ALG_SHA1,
+	/**  ODP_COMP_HASH_ALG_SHA256*/
+	ODP_COMP_HASH_ALG_SHA256
+} odp_comp_hash_alg_t;
+
+/**
+ * Comp API compression algorithm
+ *
+ */
+typedef enum {
+	/** No algorithm specified.
+	 * Means no compression, output == input.
+	 * if provided, no operation (compression/decompression or hash)
+	 * applied on input. Added for testing purpose.
+	 */
+	ODP_COMP_ALG_NULL,
+	/** DEFLATE - RFC1951 */
+	ODP_COMP_ALG_DEFLATE,
+	/** ZLIB - RFC1950 */
+	ODP_COMP_ALG_ZLIB,
+	/** LZS */
+	ODP_COMP_ALG_LZS
+} odp_comp_alg_t;
+
+/**
+ * Comp API session creation return code
+ *
+ */
+typedef enum {
+	/** Session created */
+	ODP_COMP_SES_CREATE_ERR_NONE,
+	/** Creation failed, no resources */
+	ODP_COMP_SES_CREATE_ERR_ENOMEM,
+	/** Creation failed, bad compression params */
+	ODP_COMP_SES_CREATE_ERR_INV_COMP,
+	/** Creation failed, bad hash params */
+	ODP_COMP_SES_CREATE_ERR_INV_HASH,
+	/** Creation failed,requested configuration not supported*/
+	ODP_COMP_SES_CREATE_ERR_NOT_SUPPORTED
+} odp_comp_ses_create_err_t;
+
+/**
+ * Comp API operation return codes
+ *
+ */
+typedef enum {
+	/** Operation completed successfully*/
+	ODP_COMP_ERR_NONE,
+	/** Invalid user data pointers*/
+	ODP_COMP_ERR_DATA_PTR,
+	/** Invalid input data size*/
+	ODP_COMP_ERR_DATA_SIZE,
+	/**  Compression and/or hash Algo failure*/
+	ODP_COMP_ERR_ALGO_FAIL,
+	/** Operation paused due to insufficient output buffer.
+	*
+	* This is not an error condition. On seeing this situation,
+	* Implementation should maintain context of in-progress operation and
+	* application should call packet processing API again with valid
+	* output buffer but no other alteration to operation params
+	* (odp_comp_op_param_t).
+	*
+	* if using async mode, application should either make sure to
+	* provide sufficient output buffer size OR maintain relevant
+	* context (or ordering) information with respect to each input packet
+	* en-queued for processing.
+	*
+	*/
+	ODP_COMP_ERR_OUT_OF_SPACE,
```

Comment:
You mean after ODP_COMP_ERR_NONE? such as
typedef enum{
ODP_COMP_ERR_NONE,
ODP_COMP_ERR_OUT_OF_SPACE,
	/** Invalid user data pointers*/
	ODP_COMP_ERR_DATA_PTR,
	/** Invalid input data size*/
	ODP_COMP_ERR_DATA_SIZE,
	/**  Compression and/or hash Algo failure*/
..and so on... ?